Sources/WebSocketKit/PMCE.swift

```diff
@@ -0,0 +1,753 @@
+import NIOHTTP1
+import NIOWebSocket
+import CompressNIO
+import NIO
+import Foundation
+import NIOCore
+import NIOConcurrencyHelpers
+import Logging
+
+/// The PMCE class provides methods for exchanging compressed and decompressed frames following RFC 7692.
+public final class PMCE: Sendable {
```

Comment:
Idk maybe it could be internal. At one point I was trying to make it a separate package but that seemed ultimately like too much trouble. Making it public would make it easier to use outside of vapor ? That was an original req for me but since there is no product owner.